tests/unittests/portadmin/portadmin_test.py

```diff
@@ -1,152 +1,129 @@
 from mock import Mock
 
-import unittest
+import pytest
 
 from nav.oids import OID
 from nav.enterprise.ids import VENDOR_ID_HEWLETT_PACKARD, VENDOR_ID_CISCOSYSTEMS
-from nav.portadmin.management import *
-
-###############################################################################
-from nav.portadmin.vlan import FantasyVlan
-
-
-class PortadminResponseTest(unittest.TestCase):
-    def setUp(self):
-        self.profile = Mock()
-        self.profile.snmp_version = 2
-        self.profile.snmp_community = "public"
-
-        self.hpVendor = Mock()
-        self.hpVendor.id = u'hp'
-
-        self.ciscoVendor = Mock()
-        self.ciscoVendor.id = u'cisco'
-
-        self.hpType = Mock()
-        self.hpType.vendor = self.hpVendor
-        self.hpType.get_enterprise_id.return_value = VENDOR_ID_HEWLETT_PACKARD
-
-        self.ciscoType = Mock()
-        self.ciscoType.vendor = self.ciscoVendor
-        self.ciscoType.get_enterprise_id.return_value = VENDOR_ID_CISCOSYSTEMS
-
-        self.netboxHP = Mock()
-        self.netboxHP.type = self.hpType
-        self.netboxHP.ip = '10.240.160.39'
-        self.netboxHP.get_preferred_snmp_management_profile.return_value = self.profile
-
-        self.netboxCisco = Mock()
-        self.netboxCisco.type = self.ciscoType
-        self.netboxCisco.ip = '10.240.160.38'
-        self.netboxCisco.get_preferred_snmp_management_profile.return_value = (
-            self.profile
-        )
-
-        self.snmpReadOnlyHandler = None
-        self.handler = None
-
-    def tearDown(self):
-        self.hpVendor = None
-        self.ciscoVendor = None
-        self.hpType = None
-        self.ciscoType = None
-        self.netboxHP = None
-        self.netboxCisco = None
-        self.handler = None
-        self.snmpReadOnlyHandler = None
-        self.snmpReadWriteHandler = None
-
-    ####################################################################
-    #  HP-netbox
-
-    def test_management_factory_get_hp(self):
-        self.handler = ManagementFactory.get_instance(self.netboxHP)
-        self.assertNotEqual(self.handler, None, 'Could not get handler-object')
-        self.assertIsInstance(self.handler, HP, msg='Wrong handler-type')
-
-    def test_get_vlan_hp(self):
-        self.handler = ManagementFactory.get_instance(self.netboxHP)
+from nav.portadmin.management import ManagementFactory
+from nav.portadmin.snmp.hp import HP
+from nav.portadmin.snmp.cisco import Cisco
+
+
+class TestPortadminManagementFactory:
+    def test_get_hp(self, netbox_hp):
+        handler = ManagementFactory.get_instance(netbox_hp)
+        assert handler != None, "Could not get handler-object"
```

Comment:
```suggestion
        assert handler is not None, "Could not get handler-object"
```

Never use equality tests with `None` values :)

Reply:
fixd

Reply:
fixd